src/services/hooks.rs

```diff
@@ -1,18 +1,27 @@
+use crate::cache::cache::Cache;
+use crate::cache::cache_operations::Invalidate;
 use crate::models::backend::webhooks::{Payload, PayloadDetails};
-use crate::utils::cache::{Cache, CacheExt};
 use crate::utils::errors::ApiResult;
 
 pub fn invalidate_caches(cache: &impl Cache, payload: &Payload) -> ApiResult<()> {
-    cache.invalidate_caches(&payload.address);
+    Invalidate::new()
+        .pattern(format!("c_re*{}*", &payload.address))
```

Comment:
I don't like to much, that we have to expose this prefix. Could we have this somehow as part of the builder?

Reply:
I was thinking the same. Would you oppose to having an enum for this? Something like 

```rust
pub enum InvalidationPatterns{
   requests_only, responses_only, requests_and_response, tokens, flush_all
}
```

I would add variants as we need them, but for now we would only need `requests_and_responses` and `flush_all`

Reply:
We could also have a variant `Custom(String)` that allows passing a value or a pattern

Reply:
Makes sense (please upper case the enums :D). I would probably start out only with the ones we need ;) but yea long run we can then add more